test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -950,8 +951,16 @@ private Object runExtensionMethod(ReflectiveInvocationContext<Method> invocation
             // because the test method runs from a class loaded from the TCCL
             List<Object> originalArguments = invocationContext.getArguments();
             List<Object> argumentsFromTccl = new ArrayList<>();
+            String patternString = runningQuarkusApplication.getConfigValue("quarkus.test.class-clone-pattern", String.class)
+                    .orElse("java.*");
+            Pattern clonePattern = Pattern.compile(patternString);
```

Comment:
Wondering if we should rather compile the pattern once globally rather than for each method?